src/client/options/mod.rs

```diff
@@ -588,7 +588,7 @@ impl ServerCertVerifier for NoCertVerifier {
 
 impl TlsOptions {
     /// Converts `TlsOptions` into a rustls::ClientConfig.
-    pub fn into_rustls_config(self) -> Result<rustls::ClientConfig> {
+    pub(crate) fn into_rustls_config(self) -> Result<rustls::ClientConfig> {
```

Comment:
I think this is mostly useful for us rather than users, so it seems safe to just internalize.